### PR TITLE
protein-translation: input validation test cases

### PR DESCRIPTION
**Pull request by SleeplessByte**
Adds a few test cases for input validation. JavaScript has had some of these for a long time.

FYI, I also considered grouping the cases like JavaScript, but I think the gain is minimal but churn high, so I didn't. Should someone check out this commentary, this is how we've done it in JavaScript:

```text
√ Empty RNA has no proteins (5 ms)
    Single codons
      √ Methionine RNA sequence 1 translates into Methionine (1 ms)
      √ Phenylalanine RNA sequence 1 translates into Phenylalanine (1 ms)
      √ Phenylalanine RNA sequence 2 translates into Phenylalanine (1 ms)
      √ Leucine RNA sequence 1 translates into Leucine (1 ms)
      √ Leucine RNA sequence 2 translates into Leucine
      √ Serine RNA sequence 1 translates into Serine (1 ms)
      √ Serine RNA sequence 2 translates into Serine
      √ Serine RNA sequence 3 translates into Serine
      √ Serine RNA sequence 4 translates into Serine
      √ Tyrosine RNA sequence 1 translates into Tyrosine
      √ Tyrosine RNA sequence 2 translates into Tyrosine (1 ms)
      √ Cysteine RNA sequence 1 translates into Cysteine
      √ Cysteine RNA sequence 2 translates into Cysteine
      √ Tryptophan RNA sequence 1 translates into Tryptophan (1 ms)
      √ STOP codon RNA sequence 1
      √ STOP codon RNA sequence 2 (1 ms)
      √ STOP codon RNA sequence 3
    Multiple codons
      √ Sequence of two protein codons translates into proteins (1 ms)
      √ Sequence of two different protein codons translates into proteins
      √ Translate RNA strand into correct protein list (1 ms)
      √ Translation stops if STOP codon at beginning of sequence
      √ Translation stops if STOP codon at end of three-codon sequence
      √ Translation stops if STOP codon in middle of three-codon sequence
      √ Translation stops if STOP codon in middle of six-codon sequence
    Unexpected strands
      √ Non-existing codon can't translate (6 ms)
      √ Unknown amino acids, not part of a codon, can't translate (2 ms)
      √ Incomplete RNA sequence can't translate (3 ms)
      √ Incomplete RNA sequence can translate if valid until a STOP codon
```

For the person merging: Do **NOT** squash.

### CODE REVIEW

File: exercises/protein-translation/canonical-data.json
Comment (ErikSchierboom) [inline exercises/protein-translation/canonical-data.json:222]: Shouldn't these be different?

File: exercises/protein-translation/canonical-data.json
Comment (ErikSchierboom) [inline exercises/protein-translation/canonical-data.json:295]: This is the first time an error case has been introduced to this exercise. I'm fine with that, but I did open up https://github.com/exercism/problem-specifications/pull/1758 to possibly annotate error test cases with a specific scenario for easy filtering.

**Comment by ee7** [comment]
> For the person merging: Do **NOT** squash.

I agree that the "format" and "add test cases" commits should stay separate.

But just to make sure that this PR doesn't get merged as-is: right now, this PR shouldn't be merged via the "Rebase and merge" button. We should first rewrite the current 2nd commit (https://github.com/exercism/problem-specifications/pull/1756/commits/dacf1668fdc95afa25752bbfa44c9e8138543509) to include the current 3rd commit (https://github.com/exercism/problem-specifications/pull/1756/commits/2bb374c44e2e5565db543f66f535e28e6916edf2). While we're at it, let's also rewrite the commit messages to start with `protein-translation`?

At first I wondered why a PR that says "do not squash" included a commit called "Fix duplicated uuid", since I'm not aware of any duplicate UUIDs in the `problem-specifications` repo. To others: that commit just fixes a new UUID in this PR.

(I haven't reviewed this PR otherwise).

File: exercises/protein-translation/canonical-data.json
Comment (SleeplessByte) [reply exercises/protein-translation/canonical-data.json:295]: I expected to see input validation or error scenario in plenty other test cases. My suggestion would be: YES, add it.

File: exercises/protein-translation/canonical-data.json
Comment (SleeplessByte) [reply exercises/protein-translation/canonical-data.json:222]: They should not but the name isn't clear.

Comment (SleeplessByte) [reply exercises/protein-translation/canonical-data.json:222]: (The _codon_ is different, the _protein_ is not)

**Review by petertseng — APPROVED** [review]
the test cases added are correct; I do recommend rewriting the commits appropriately before merging if the guidance is going to be not to squash.

**Comment by SleeplessByte** [comment]
This is now dependent on #1758. Should be rebased once that's merged.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ee7** [comment]
Is the new first commit (https://github.com/exercism/problem-specifications/pull/1756/commits/726a6c8ed38f0c89abe9eb7ece2c5d5fe8995ab7) supposed to be style-only changes? It has the commit message "Format", but it looks like it adds two new test cases.

**Comment by SleeplessByte** [comment]
Yeah I don't know what happened there :/. I think it should just be squashed now.

**Comment by SleeplessByte** [comment]
@exercism/reviewers wanna check this out?

**Review by angelikatyborska — APPROVED** [review]
I'm loving those new test cases!

@SleeplessByte conflict resolution is needed before merging

**Comment by SleeplessByte** [comment]
Meh, the entire file is conflicted :(

**Comment by ErikSchierboom** [comment]
@SleeplessByte Are you still interested in working on this PR?

**Comment by SleeplessByte** [comment]
Eh. Yes, but I can't find the headspace to re-do the work. Wanna pick it up?

**Comment by ErikSchierboom** [comment]
@SleeplessByte I've fixed the merge conflict. Could you double-check its contents? Note: for ease of reviewing, I've left out the formatting. We can do that in a separate PR if needed.

**Review by SleeplessByte — COMMENTED** [review]
Looks good now. Feel free to format.

**Comment by ErikSchierboom** [comment]
I'll format in a different PR. Want to merge this one?

File: exercises/protein-translation/canonical-data.json
Comment (wolf99) [inline exercises/protein-translation/canonical-data.json:264]:
```suggestion
      "description": "Unknown amino acids, not part of a codon, results in an error",
```

File: exercises/protein-translation/canonical-data.json
Comment (wolf99) [inline exercises/protein-translation/canonical-data.json:282]:
```suggestion
      "description": "Incomplete RNA sequence results in an error",
```

**Review by wolf99 — CHANGES_REQUESTED** [review]
Minor change to the description suggested.
I think the optionality of the test is explained well in the comments, so no need to try to indicate it in the description also.

File: exercises/protein-translation/canonical-data.json
Comment (kotp) [reply exercises/protein-translation/canonical-data.json:282]: I think "can't translate" gives the reason why it errors rather than a more generic "results in an error" which we might already know if there is an error raised.

Comment (ErikSchierboom) [reply exercises/protein-translation/canonical-data.json:282]: I agree with kotp here.

Comment (SleeplessByte) [reply exercises/protein-translation/canonical-data.json:282]: That's why I didn't say "results in an error". I expected maintainers to use the appends to get the behavior described.

Comment (wolf99) [reply exercises/protein-translation/canonical-data.json:282]: Seems like the consensus is against me here 😄.
I still think I'm correct, but I don't think it's a big enough issue to hold up the PR, so I remove my request for change 👍

**Review by wolf99 — APPROVED** [review]

File: exercises/protein-translation/canonical-data.json
Comment (kotp) [reply exercises/protein-translation/canonical-data.json:282]: To be fair, you are not wrong, and if it were for a single track, then I could recommend doing it that way.

The comments that follow indicates that it may not raise an error, and so it would be in conflict with the comment.  It is up to the track to decide what to do with 'can't translate' and while raising an error is indeed one correct way to do this, it is not the only correct way.

Comment (SleeplessByte) [reply exercises/protein-translation/canonical-data.json:282]: I agree. You're not wrong, but it's the problem of writing this for tracks that don't even have errors.

**Pull request merged**
